File: server/src/main/java/com/nowakArtur97/globalTerrorismAPI/feature/event/EventModel.java
package com.nowakArtur97.globalTerrorismAPI.feature.event;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.nowakArtur97.globalTerrorismAPI.feature.city.CityModel;
import com.nowakArtur97.globalTerrorismAPI.feature.target.TargetModel;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.*;
import org.springframework.hateoas.RepresentationModel;

import java.util.Date;
import java.util.Objects;

@ApiModel(description = "Details about the Target")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class EventModel extends RepresentationModel<EventModel> implements Event {

    @ApiModelProperty(notes = "The unique id of the Event")
    private Long id;

    @ApiModelProperty(notes = "The event's summary")
    private String summary;

    @ApiModelProperty(notes = "The event's motive")
    private String motive;

    @ApiModelProperty(notes = "The event's date")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private Date date;

    @ApiModelProperty(notes = "Was the event part of multiple incidents?")
    private Boolean isPartOfMultipleIncidents;

    @ApiModelProperty(notes = "Was the event successful?")
    private Boolean isSuccessful;

    @ApiModelProperty(notes = "Was the event suicide?")
    private Boolean isSuicidal;

    @ApiModelProperty(notes = "The event's target")
    private TargetModel target;

    @ApiModelProperty(notes = "The event's city")
    private CityModel city;

    @Override
    public boolean equals(Object o) {

        if (this == o) return true;
        if (!(o instanceof EventModel)) return false;

        EventModel that = (EventModel) o;
        return Objects.equals(getId(), that.getId()) &&
                Objects.equals(getSummary(), that.getSummary()) &&
                Objects.equals(getMotive(), that.getMotive()) &&
                Objects.equals(getDate(), that.getDate()) &&
                Objects.equals(getIsPartOfMultipleIncidents(), that.getIsPartOfMultipleIncidents()) &&
                Objects.equals(getIsSuccessful(), that.getIsSuccessful()) &&
                Objects.equals(getIsSuicidal(), that.getIsSuicidal());
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), getId(), getSummary(), getMotive(), getDate(),
                getIsPartOfMultipleIncidents(), getIsSuccessful(), getIsSuicidal());
    }
}

File: client/src/app/events/models/event.model.ts
import City from 'src/app/cities/models/city.model';

export default class Event {
  constructor(
    private readonly _id: number,
    private _summary: string,
    private _motive: string,
    private _date: Date,
    private _isPartOfMultipleIncidents: boolean,
    private _isSuccessful: boolean,
    private _isSuicidal: boolean,
    private _city: City
  ) {}

  public get id(): number {
    return this._id;
  }

  public get summary(): string {
    return this._summary;
  }

  public get motive(): string {
    return this._motive;
  }

  public get date(): Date {
    return this._date;
  }

  public get isPartOfMultipleIncidents(): boolean {
    return this._isPartOfMultipleIncidents;
  }

  public get isSuccessful(): boolean {
    return this._isSuccessful;
  }

  public get isSuicidal(): boolean {
    return this._isSuicidal;
  }

  public get city(): City {
    return this._city;
  }
}
